src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java

```diff
@@ -0,0 +1,128 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+
+@RestController
+@Api(value = "/pregnanttreatmenttypes", produces = MediaType.APPLICATION_JSON_VALUE)
+public class PregnantTreatmentTypeController {
+
+	@Autowired
+	protected PregnantTreatmentTypeBrowserManager pregTreatTypeManager;
+	
+	@Autowired
+	protected PregnantTreatmentTypeMapper mapper;
+
+	private final Logger logger = LoggerFactory.getLogger(PregnantTreatmentTypeController.class);
+
+	public PregnantTreatmentTypeController(PregnantTreatmentTypeBrowserManager pregTreatTypeManager, PregnantTreatmentTypeMapper pregnantTreatmentTypemapper) {
+		this.pregTreatTypeManager = pregTreatTypeManager;
+		this.mapper = pregnantTreatmentTypemapper;
+	}
+
+	/**
+	 * create a new {@link PregnantTreatmentType}
+	 * @param pregnantTreatmentTypeDTO
+	 * @return <code>true</code> if the pregnant treatment type has been stored, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PostMapping(value = "/pregnanttreatmenttypes", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> newPregnantTreatmentType(@RequestBody PregnantTreatmentTypeDTO pregnantTreatmentTypeDTO) throws OHServiceException {
+		String code = pregnantTreatmentTypeDTO.getCode();
+		logger.info("Create pregnant treatment Type " + code);
+		boolean isCreated = pregTreatTypeManager.newPregnantTreatmentType(mapper.map2Model(pregnantTreatmentTypeDTO));
+		PregnantTreatmentType pregTreatTypeCreated = null;
+		List<PregnantTreatmentType> pregTreatTypeFounds = pregTreatTypeManager.getPregnantTreatmentType().stream().filter(pregtreattype -> pregtreattype.getCode().equals(code))
+				.collect(Collectors.toList());
+		if (pregTreatTypeFounds.size() > 0)
+			pregTreatTypeCreated = pregTreatTypeFounds.get(0);
+		if (!isCreated || pregTreatTypeCreated == null) {
+			throw new OHAPIException(new OHExceptionMessage(null, "pregnant treatment Type is not created!", OHSeverityLevel.ERROR));
+		}
+		return ResponseEntity.status(HttpStatus.CREATED).body(pregTreatTypeCreated.getCode());
+	}
+
+	/**
+	 * Updates the specified {@link PregnantTreatmentType}.
+	 * @param pregnantTreatmentTypeDTO
+	 * @return <code>true</code> if the pregnant treatment type has been updated, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/pregnanttreatmenttypes", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> updatePregnantTreatmentTypet(@RequestBody PregnantTreatmentTypeDTO pregnantTreatmentTypeDTO)
```

Comment:
Put Method should have code as @PathVariable. I know it should be an http standard.
`@PutMapping(value = "/pregnanttreatmenttypes/{code}", produces = MediaType.APPLICATION_JSON_VALUE)`

Reply:
the code is already in the dto, should I add it again as @PathVariable ?